front_end/content.py

```diff
@@ -330,6 +379,41 @@ def get_user_count(self):
         self.cursor.execute(sql)
         return self.cursor.fetchone()["count"]
 
+    def check_course_exists(self, course_id):
+        sql = '''SELECT COUNT(*) AS count
+                 FROM courses
+                 WHERE course_id = ?'''
+        self.cursor.execute(sql, (course_id,))
+        if self.cursor.fetchone():
+            return True
+        else:
+            return False
+
+    def get_course_from_role(self, user_id):
+        courses = []
+        sql = '''SELECT p.course_id, c.title
+                 FROM permissions p
+                 INNER JOIN courses c
+                 ON p.course_id = c.course_id
+                 WHERE user_id = ?'''
+
+        self.cursor.execute(sql, (user_id,))
+        for course in self.cursor.fetchall():
+            course_basics = {"id": course["course_id"], "title": course["title"]}
+            courses.append([course["course_id"], course_basics])
+        return courses
+
+    def get_course_passcode(self, course_id):
```

Comment:
I would rather that the passcode be returned with course details rather than having a separate function for returning just the passcode. Although just returning the passcode is a bit faster, I'm concerned about keeping the code as simple as possible.